packages/nextjs/test/config/wrappers.test.ts

```diff
@@ -0,0 +1,70 @@
+import * as SentryCore from '@sentry/core';
+import * as SentryTracing from '@sentry/tracing';
+import { IncomingMessage, ServerResponse } from 'http';
+
+import {
+  withSentryGetServerSideProps,
+  withSentryServerSideGetInitialProps,
+  // TODO: Leaving `withSentryGetStaticProps` out for now until we figure out what to do with it
+  // withSentryGetStaticProps,
+  // TODO: Leaving these out for now until we figure out pages with no data fetchers
+  // withSentryServerSideAppGetInitialProps,
+  // withSentryServerSideDocumentGetInitialProps,
+  // withSentryServerSideErrorGetInitialProps,
+} from '../../src/config/wrappers';
+
+const startTransactionSpy = jest.spyOn(SentryCore, 'startTransaction');
+const setMetadataSpy = jest.spyOn(SentryTracing.Transaction.prototype, 'setMetadata');
+
+describe('data-fetching function wrappers', () => {
+  const route = '/tricks/[trickName]';
+  let req: IncomingMessage;
+  let res: ServerResponse;
+
+  describe('starts a transaction if tracing enabled', () => {
+    beforeEach(() => {
+      req = { headers: {}, url: 'http://dogs.are.great/tricks/kangaroo' } as IncomingMessage;
+      res = {} as ServerResponse;
```

Comment:
Any reason why we can't define those variables for each test individually? That would side-effects between tests at a minimum.

Reply:
Tests run serially within a given test file I believe, though, right? So resetting them before each test should in theory prevent any side effects. (And doing it this way saves having the boilerplate at the beginning of every test. On that topic, I actually tried doing this as a `test.each` but IIRC I couldn't get the types to work correctly.)

Reply:
Yeah but why depend on jest behavior when you can simply make something be scoped to a single test? It is also easier to read and debug imo since you don't have to jump around to a globally defined variable and check whether it's overwritten at some point by some other test or if it is reset on some `afterEach` hook.

I don't want to turn these tests into `test.each`, I just want to remove the mutability of test inputs. I am not so worried about this particular file since it is very small but I'd like to avoid this pattern in other places. That's why I am bringing it up here.